src/main/kotlin/com/criptext/mail/scenes/mailbox/data/EmailInsertionSetup.kt

```diff
@@ -245,6 +245,30 @@ object  EmailInsertionSetup {
             body
     }
 
+    fun getDecryptedEmailHeaders(signalClient: SignalClient,
+                              body: String,
+                              metadata: EmailMetadata): String? {
+        if(metadata.senderRecipientId == SignalUtils.externalRecipientId
+                || metadata.isExternal == true) return null
+        val senderId = getSenderId(metadata)
+        return if (metadata.messageType != null && senderId.first != null) {
+            val encryptedData = SignalEncryptedData(
+                    encryptedB64 = body,
+                    type = metadata.messageType)
+
+            try {
+                decryptMessage(signalClient = signalClient,
+                        recipientId = senderId.second, deviceId = senderId.first!!,
+                        encryptedData = encryptedData,
+                        isFromBob = metadata.isExternal == true
+                                && metadata.senderRecipientId == SignalUtils.externalRecipientId)
+            } catch (e: Exception){
+                null
+            }
+        } else
+            null
```

Comment:
```suggestion
            body
```

Reply:
Since this is the headers, the default value es null.

Reply:
what If the emails comes unencrypted?